File: src/models/llm_model.go
package models

import (
	"context"
	"discord-gemini-bot/src/types"
)

// LLMModel is an abstract interface for Large Language Models
type LLMModel interface {
	// GenerateAsync generates text asynchronously based on the given prompt
	GenerateAsync(ctx context.Context, prompt string, images []map[string]interface{}) (string, error)
	
	// GenerateWithHistoryAsync generates text asynchronously with conversation history
	GenerateWithHistoryAsync(ctx context.Context, messages []*types.Message) (string, error)
	
	// SetSystemPrompt sets the system prompt for the model
	SetSystemPrompt(systemPrompt string)
}

File: python_deprecated/src/models/llm_model.py
from abc import ABC, abstractmethod
from typing import List, Optional, Dict, Any

class LLMModel(ABC):
    """Abstract base class for Large Language Models"""
    
    @abstractmethod
    async def generate_async(self, prompt: str, **kwargs) -> str:
        """Generate text asynchronously based on the given prompt"""
        pass
    
    @abstractmethod
    async def generate_with_history_async(self, messages: List[Dict[str, str]], **kwargs) -> str:
        """Generate text asynchronously with conversation history"""
        pass
    
    @abstractmethod
    def set_system_prompt(self, system_prompt: str) -> None:
        """Set the system prompt for the model"""
        pass
    